test/e2e/case28_trigger_uninstall_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package e2e
+
+import (
+	"context"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"open-cluster-management.io/config-policy-controller/pkg/common"
+	"open-cluster-management.io/config-policy-controller/pkg/triggeruninstall"
+	"open-cluster-management.io/config-policy-controller/test/utils"
+)
+
+// This test only works when the controller is running in the cluster.
+var _ = Describe("Clean up during uninstalls", Label("running-in-cluster"), Ordered, func() {
+	const (
+		configMapName        string = "case29-trigger-uninstall"
+		deploymentName       string = "config-policy-controller"
+		deploymentNamespace  string = "open-cluster-management-agent-addon"
+		policyName           string = "case29-trigger-uninstall"
+		policy2Name          string = "case29-trigger-uninstall2"
+		policyYAMLPath       string = "../resources/case29_trigger_uninstall/policy.yaml"
+		policy2YAMLPath      string = "../resources/case29_trigger_uninstall/policy2.yaml"
+		pruneObjectFinalizer string = "policy.open-cluster-management.io/delete-related-objects"
+	)
+
+	It("verifies that finalizers are removed when being uninstalled", func() {
+		By("Creating two configuration policies with pruneObjectBehavior")
+		utils.Kubectl("apply", "-f", policyYAMLPath, "-n", testNamespace)
+		utils.Kubectl("apply", "-f", policy2YAMLPath, "-n", testNamespace)
+
+		By("Verifying that the configuration policies are compliant and have finalizers")
+		Eventually(func(g Gomega) {
+			policy := utils.GetWithTimeout(
+				clientManagedDynamic, gvrConfigPolicy, policyName, testNamespace, true, defaultTimeoutSeconds,
+			)
+			g.Expect(utils.GetComplianceState(policy)).To(Equal("Compliant"))
+
+			g.Expect(policy.GetFinalizers()).To(ContainElement(pruneObjectFinalizer))
+		}, defaultTimeoutSeconds, 1).Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			policy2 := utils.GetWithTimeout(
+				clientManagedDynamic, gvrConfigPolicy, policy2Name, testNamespace, true, defaultTimeoutSeconds,
+			)
+			g.Expect(utils.GetComplianceState(policy2)).To(Equal("Compliant"))
+
+			g.Expect(policy2.GetFinalizers()).To(ContainElement(pruneObjectFinalizer))
+		}, defaultTimeoutSeconds, 1).Should(Succeed())
+
+		By("Triggering an uninstall")
+		kubeconfigBytes, err := os.ReadFile(kubeconfigManaged)
+		Expect(err).To(BeNil())
+
+		config, err := clientcmd.RESTConfigFromKubeConfig(kubeconfigBytes)
+		Expect(err).To(BeNil())
+
+		ctx, ctxCancel := context.WithDeadline(
+			context.Background(),
+			// Cancel the context after the default timeout seconds to cause TriggerUninstall to stop prematurely
+			time.Now().Add(time.Duration(defaultTimeoutSeconds)*time.Second),
```

Comment:
By cancelling the context automatically, this does not verify that TriggerUninstall would eventually complete on its own. Was there a problem with waiting for it?

Reply:
This was to avoid the test running forever if `TriggerUninstall` misbehaves. I realize the comment is not clear. I'll update that.